From 9e263bf3571cf57e8dd111faf84fd0176694ebff Mon Sep 17 00:00:00 2001
From: Ming Ma <mingma@apache.org>
Date: Wed, 22 Feb 2017 09:41:07 -0800
Subject: [PATCH 2314/2848] HDFS-11411. Avoid OutOfMemoryError in
 TestMaintenanceState test runs. (Manoj
 Govindassamy via mingma)

(cherry picked from commit cfcd527323352cf2a851c5c41f5d243d375d88d0)
(cherry picked from commit 0add328c8c1d86f08793682d7a37a7b30509e28b)

Change-Id: I02217bd0f00d7a473f98f81ff12fd498e761823e
---
 .../apache/hadoop/hdfs/TestMaintenanceState.java   |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestMaintenanceState.java b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestMaintenanceState.java
index 9cc130b..ba1fd1b 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestMaintenanceState.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/test/java/org/apache/hadoop/hdfs/TestMaintenanceState.java
@@ -332,6 +332,7 @@ private void testExpectedReplication(int replicationFactor)
 
   private void testExpectedReplication(int replicationFactor,
       int expectedReplicasInRead) throws IOException {
+    setup();
     startCluster(1, 5);
 
     final Path file = new Path("/testExpectedReplication.dat");
@@ -351,6 +352,7 @@ private void testExpectedReplication(int replicationFactor,
         nodeOutofService));
 
     cleanupFile(fileSys, file);
+    teardown();
   }
 
   /**
@@ -491,6 +493,7 @@ public void testDecommissionDifferentNodeAfterMaintenances()
 
   private void testDecommissionDifferentNodeAfterMaintenance(int repl)
       throws Exception {
+    setup();
     startCluster(1, 5);
 
     final Path file =
@@ -518,6 +521,7 @@ private void testDecommissionDifferentNodeAfterMaintenance(int repl)
     assertNull(checkWithRetry(ns, fileSys, file, repl + 1, null));
 
     cleanupFile(fileSys, file);
+    teardown();
   }
 
 
@@ -547,6 +551,7 @@ public void testChangeReplicationFactors() throws IOException {
    */
   private void testChangeReplicationFactor(int oldFactor, int newFactor,
       int expectedLiveReplicas) throws IOException {
+    setup();
     LOG.info("Starting testChangeReplicationFactor {} {} {}",
         oldFactor, newFactor, expectedLiveReplicas);
     startCluster(1, 5);
@@ -579,6 +584,7 @@ private void testChangeReplicationFactor(int oldFactor, int newFactor,
     assertNull(checkWithRetry(ns, fileSys, file, newFactor, null));
 
     cleanupFile(fileSys, file);
+    teardown();
   }
 
 
-- 
1.7.9.5

